core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -198,18 +167,33 @@ Map<String, String> params() {
   }
 
   public RESTSessionCatalog() {
-    this(config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build(), null);
+    this(
+        config -> {
+          HTTPClient.Builder builder =
+              HTTPClient.builder(config).uri(config.get(CatalogProperties.URI));
+          configHeaders(config).forEach(builder::withHeader);
+          return builder.build();
+        },
+        null);
   }
 
   public RESTSessionCatalog(
       Function<Map<String, String>, RESTClient> clientBuilder,
       BiFunction<SessionContext, Map<String, String>, FileIO> ioBuilder) {
+    this(clientBuilder, ioBuilder, AuthManagers::loadAuthManager);
+  }
+
+  public RESTSessionCatalog(
+      Function<Map<String, String>, RESTClient> clientBuilder,
+      BiFunction<SessionContext, Map<String, String>, FileIO> ioBuilder,
+      BiFunction<String, Map<String, String>, AuthManager> authManagerBuilder) {
```

Comment:
Is there a reason we need to expose this?  I really don't love tracking a `BiFunction<...>` as a member reference and this is adding something that I don't think is currently exposed.  I don't see anywhere this is actually used either, so it seems we could follow up if there's a use case, but for now, we can just encapsulate it.

Reply:
Yes, I introduced this for our Trino friends, see this comment for context:

https://github.com/apache/iceberg/pull/12197#issuecomment-2678010968

TLDR, Trino would like to create catalogs with an externally-provided AuthManager. I think the use case is valid.

But OK, let's remove for now and reassess later. They can still use reflection to create their AuthManager.

\cc @varpa89

Reply:
Removed.

Reply:
I'm open to discussing how to support use cases like this, but I don't think this is the right way to go about it.  Passing around functional references with little context isn't a great interface.  Let's take it out for now and figure out how to do this in a cleaner and more consistent way.